include/line_numbers.h

```diff
@@ -0,0 +1,52 @@
+#ifndef LINE_NUMBERS_H
+#define LINE_NUMBERS_H
+
+// The macros in this file serve to reduce clutter from version differences that are only due to line numbers being different.
+
+/**
+ * Macro that defines line numbers by version.
+ * This macro is not meant to be used externally, see wrappers below.
+ */
+#define LN(ntsc_1_0, ntsc_1_1, ntsc_1_2, pal_1_0, pal_1_1, gc_jp, gc_jp_mq, gc_us, gc_us_mq, gc_eu, gc_eu_mq, gc_jp_ce, debug) \
+    (OOT_VERSION == NTSC_1_0) ? ntsc_1_0 : \
+    (OOT_VERSION == NTSC_1_1) ? ntsc_1_1 : \
+    (OOT_VERSION == NTSC_1_2) ? ntsc_1_2 : \
+    (OOT_VERSION == PAL_1_0)  ? pal_1_0  : \
+    (OOT_VERSION == PAL_1_1)  ? pal_1_1  : \
+    (OOT_VERSION == GC_JP)    ? gc_jp    : \
+    (OOT_VERSION == GC_JP_MQ) ? gc_jp_mq : \
+    (OOT_VERSION == GC_US)    ? gc_us    : \
+    (OOT_VERSION == GC_US_MQ) ? gc_us_mq : \
+    (OOT_VERSION == GC_EU)    ? gc_eu    : \
+    (OOT_VERSION == GC_EU_MQ) ? gc_eu_mq : \
+    (OOT_VERSION == GC_JP_CE) ? gc_jp_ce : debug
+
+/**
+ * These wrappers exist to remove duplication of line numbers being passed to LN.
+ * The wrappers have ambiguous names purposefully so that they are short and don't add clutter.
+ *
+ * The groupings of different line numbers are difficult to name, as the exact grouping
+ * depends on which versions happen to have changes in a given file.
+ */
+
+// NTSC 1.0 and all the other versions
+#define LN1(ntsc_1_0, other) \
+    LN(ntsc_1_0, other, other, other, other, other, other, other, other, other, other, other, other)
+
+// NTSC 1.0, NTSC 1.1, and all the other versions
+#define LN2(ntsc_1_0, ntsc_1_1, other) \
+    LN(ntsc_1_0, ntsc_1_1, other, other, other, other, other, other, other, other, other, other, other)
+
+// NTSC 1.0, Gamecube, and all the other versions
+#define LN3(ntsc_1_0, other, gc) \
+    LN(ntsc_1_0, other, other, other, other, gc, gc, gc, gc, gc, gc, gc, gc)
+
+// NTSC 1.0, NTSC 1.1, Gamecube, and all the other versions
+#define LN4(ntsc_1_0, ntsc_1_1, other, gc) \
+    LN(ntsc_1_0, ntsc_1_1, other, other, other, gc, gc, gc, gc, gc, gc, gc, gc)
+
+// NTSC 1.0, NTSC 1.1, NTSC 1.2/PAL 1.0, Gamecube US/JP, Gamecube EU, and Gamecube CE
+#define LN5(ntsc_1_0, ntsc_1_1, other, pal_1_1, gc_ntsc, gc_eu, gc_jp_ce) \
+    LN(ntsc_1_0, ntsc_1_1, other, other, pal_1_1, gc_ntsc, gc_ntsc, gc_ntsc, gc_ntsc, gc_eu, gc_eu, gc_jp_ce, gc_eu)
```

Comment:
I find these `LNx` macros pretty inelegant. What about using `LN` directly, e.g.

```c
HUNGUP_AND_CRASH("../z_play.c",
                 LN(2263, 2266, 2269, 2269, 2272, 2287, 2287, 2287, 2287, 2290, 2290, 2290, 2293))
```

I tried a couple places and it fits on a single line so I don't think the formatting would be too bad

Reply:
I find the duplicating unfortunate, but I will add a commit with all of them done so we can see what it looks like.

Reply:
Also note the list will get longer as we add more versions

Reply:
> Also note the list will get longer as we add more versions

yeah good point. Another option would be to add an `LNx` for anything used more than once or twice (`LN1`-`LN4`) but not if it's used only once (so no `LN5` in this case)

Reply:
because I think `LN5` is the most ugly to me, it just seems so ... ad-hoc

Reply:
I agree its arbitrary and it kinda sucks, but it serves its purpose well of reducing duplicates, even if its used only once. As mentioned the list will still get long, so the z_play case will still stick out like an eyesore. Why not just hide it?

Reply:
For example, I forsee that the current debug rom line number would work for the other two roms. Both iques can probably be deduplicated. It will still help reduce line length in this one case versus using the whole LN macro

Reply:
alright, I'm okay with the way it is now